### PR TITLE
build: remove whitespace at eof and between tags

### LINKED ISSUE #1596: Remove trailing whitespace in inlined templates
#### Bug, feature request, or proposal:

Bug
#### What is the expected behavior?

The string representing a component's template should not have a whitespace as the last character.
#### What is the current behavior?

Every material component's template appends a whitespace at the end of the string in the JS transpiled file.
#### What are the steps to reproduce?

`<button md-button class="increment" (click)="increment()">A+</button>`

will be translated into :

`<button class="increment" md-button=""><span class="md-button-wrapper">A+</span> </button>`

... see the whitespace just before `</button>` ?
It will be appended before the closing tag of any material element whose template file has a new line before the EndOfFile...
#### What is the use-case or motivation for changing an existing behavior?

the following test should work :

```
it('should display a button that reads "A+" ', () => {
        const buttonContent = button.textContent;
        expect(buttonContent).toEqual('A+');
    });
```
#### Which versions of Angular, Material, OS, browsers are affected?

Any version that follows the official style guide's linting recommendations.
#### Is there anything else we should know?

You're awesome !

### PR DESCRIPTION
**Pull request by willshowell**
Fixes #1596

### CODE REVIEW

File: scripts/release/inline-resources.js [hunk 1]
```diff
@@ -93,7 +93,9 @@ function inlineTemplate(content, urlResolver) {
     const templateContent = fs.readFileSync(templateFile, 'utf-8');
     const shortenedTemplate = templateContent
       .replace(/([\n\r]\s*)+/gm, ' ')
```
Comment (DevVersion) [inline scripts/release/inline-resources.js]: This seems to be a overkill fix. Why not just remove the space here and just replace it with an empty string?

Comment (willshowell) [inline scripts/release/inline-resources.js]: Wouldn't that remove the spaces between attributes too aggressively? Such as after `<ng-content`:
```html
<ng-content select="[md-card-avatar], [mat-card-avatar]"></ng-content>
<div class="mat-card-header-text">
  <ng-content
      select="md-card-title, mat-card-title, md-card-subtitle, mat-card-subtitle,
      [md-card-title], [mat-card-title], [md-card-subtitle], [mat-card-subtitle]"></ng-content>
</div>
<ng-content></ng-content>
```

Comment (DevVersion) [inline scripts/release/inline-resources.js]: That's a good point. Couldn't we just use gulp-clean-css to do proper minifying?

Comment (willshowell) [inline scripts/release/inline-resources.js]: Yeah, sure. I considered reusing htmlmin, but that seemed like overkill to me haha. I'll close this if you'd prefer to take on a different approach.

Comment (DevVersion) [inline scripts/release/inline-resources.js]: You can just do it in that PR and you don't have to close the PR.

**Comment by jelbourn** [comment]
Not sure what the status of this PR is now- I'm just going to close it and you can open a new one if you want to go for a different approach.

**Pull request closed without merging**